polytope_server/common/queue/sqs_queue.py

```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+from . import queue
+import boto3
+import os
+import re
+from ..metric_collector import SQSQueueMetricCollector
+
+
+class SQSQueue(queue.Queue):
+    def __init__(self, config):
+        host = config.get("host")
+        queue_name = config.get("name")
+        self.keep_alive_interval = config.get("keep_alive_interval", 60)
+        self.visibility_timeout = config.get("visibility_timeout", 120)
+
+        region = self.__parse_region(host)
+        self.queue_url = "{}/{}".format(host, queue_name)
+
+        logging.getLogger("sqs").setLevel("WARNING")
+
+        self.client = boto3.client("sqs", region_name=region)
+        self.check_connection()
+        self.queue_metric_collector = SQSQueueMetricCollector(self.queue_url)
+
+    def enqueue(self, message):
+        self.client.send_message(QueueUrl=self.queue_url, MessageBody=json.dumps(message.body))
+
+    def dequeue(self):
+        response = self.client.receive_message(
+            QueueUrl=self.queue_url,
+            VisibilityTimeout=self.visibility_timeout,  # If processing takes more seconds, message will be read twice
+            MaxNumberOfMessages=1,
+        )
+        if not response["Messages"]:
+            return None
+
+        for item in response["Messages"][1:]:
+            self.client.change_message_visibility(
+                QueueUrl=self.queue_url, ReceiptHandle=item["ReceiptHandle"], VisibilityTimeout=0
+            )
+        body = response["Messages"][0]["Body"]
+        receipt_handle = response["Messages"][0]["ReceiptHandle"]
```

Comment:
```suggestion
        msg, *remainder = response["Messages"]
        for item in remainder:
            self.client.change_message_visibility(
                QueueUrl=self.queue_url, ReceiptHandle=item["ReceiptHandle"], VisibilityTimeout=0
            )
        body = msg["Body"]
        receipt_handle = msg["ReceiptHandle"]
```
The slice will fail if there is not more than 1 item

Reply:
It seems to work okay when I tested it, but I prefer your change

Reply:
Sorry, yes it does work. I was under the impression that it would raise an index error but it's not the case.